policy-controller/src/admission.rs

```diff
@@ -0,0 +1,81 @@
+use super::api::policy::{Server, ServerSpec};
+use kube::api::{Api, ListParams};
+use kube::core::{
+    admission::{AdmissionRequest, AdmissionResponse, AdmissionReview},
+    DynamicObject, ObjectList,
+};
+use std::convert::{Infallible, TryInto};
+use std::fmt;
+use tracing::{error, info, instrument};
+use warp::{reply, Reply};
+
+#[derive(Clone)]
+pub struct Admission(pub kube::Client);
+
+#[instrument]
+pub async fn handler(
+    body: AdmissionReview<DynamicObject>,
+    admission: Admission,
+) -> Result<impl Reply, Infallible> {
+    // Parse incoming webhook AdmissionRequest first
+    let req: AdmissionRequest<DynamicObject> = match body.try_into() {
+        Ok(req) => req,
+        Err(err) => {
+            error!(error = ?err.to_string(), "invalid request");
+            return Ok(reply::json(
+                &AdmissionResponse::invalid(err.to_string()).into_review(),
+            ));
+        }
+    };
+    info!(?req);
+
+    // Then construct a AdmissionResponse
+    let mut res = AdmissionResponse::from(&req);
+
+    // List existing servers.
+    let api: Api<Server> = Api::all(admission.0.clone());
```

Comment:
nit: afaict, we shouldn't have to `clone` this, since it looks like we never use `admission` anywhere else?
```suggestion
    let api: Api<Server> = Api::all(admission.0);
```